src/components/UpcomingQuizCard/UpcomingQuizCard.tsx

```diff
@@ -0,0 +1,58 @@
+import { Typography, CardContent } from '@mui/material';
+import React from 'react';
+
+import { theme } from '../../style-library';
+
+import CalendarIcon from './icon/CalendarIcon';
+import { useUpcomingQuizCard } from './UpcomingQuizCard.hook';
+import { StyledCard, DateContainer, RegisterButton } from './UpcomingQuizCard.styles';
+
+import type { UpcomingQuizCardProps } from './UpcomingQuizCard.types';
+
+export const UpcomingQuizCard: React.FC<UpcomingQuizCardProps> = ({
+  title,
+  date,
+  description,
+  onRegister,
+  ...cardProps
+}) => {
+  const { formattedDate } = useUpcomingQuizCard(date);
+
+  const getLocalDay = (dateValue: string | Date) => {
```

Comment:
getLocalDay mixes manual string-splitting and UTC-based Date creation, then calls getUTCDate() for Date objects. This is inconsistent and error-prone. Day extraction should be consistent (either local or UTC) and ideally use a date parsing library to avoid timezone/parsing bugs.

**Sugerencia:**
```typescript
Use a single robust approach: parse ISO strings with date-fns parseISO and then use getDate() on the resulting Date (local day) or format(date, 'd') to get the day string. Example:

import { parseISO } from 'date-fns';
const parsed = typeof dateValue === 'string' ? parseISO(dateValue) : dateValue;
return parsed.getDate();

This keeps behavior consistent for strings and Date objects.
```